packages/back-end/src/app/controllers/easy-genomics/organization/list-organizations.lambda.ts

```diff
@@ -2,16 +2,41 @@ import { Organization } from '@easy-genomics/shared-lib/src/app/types/easy-genom
 import { buildResponse } from '@easy-genomics/shared-lib/src/app/utils/common';
 import { APIGatewayProxyResult, APIGatewayProxyWithCognitoAuthorizerEvent, Handler } from 'aws-lambda';
 import { OrganizationService } from '@BE/services/easy-genomics/organization-service';
+import { getOrganizationAccessOrganizationIds, validateSystemAdminAccess } from '@BE/utils/auth-utils';
 
 const organizationService = new OrganizationService();
 
+/**
+ * This API is returns a list of Organizations depending on the User's Cognito
+ * Session.
+ *
+ * If the User is the SystemAdmin, then it will return all Organizations in the
+ * platform. Otherwise if it is a normal User, then it will return only the
+ * Organizations the User has been invited to.
+ *
+ * @param event
+ */
 export const handler: Handler = async (
   event: APIGatewayProxyWithCognitoAuthorizerEvent,
 ): Promise<APIGatewayProxyResult> => {
   console.log('EVENT: \n' + JSON.stringify(event, null, 2));
   try {
-    const response: Organization[] = await organizationService.list();
-    return buildResponse(200, JSON.stringify(response), event);
+    if (validateSystemAdminAccess(event)) {
+      const response: Organization[] = await organizationService.list();
+      return buildResponse(200, JSON.stringify(response), event);
+    } else {
+      const organizationIds = getOrganizationAccessOrganizationIds(event);
+      if (organizationIds.length === 0) {
+        throw new Error('Unauthorized access');
```

Comment:
by my reckoning this could be a valid case (user is removed from all organizations but can still log in and do some things) so it seems overzealous to me to return an error - rather simply returning an empty list should suffice so the website shows no organizations but otherwise it's business as usual. thoughts @zigtan ?

Reply:
@willholt-tb , that's true. I'll remove this logic check.